src/snowflake/snowpark/dataframe.py

```diff
@@ -487,10 +488,22 @@ def __init__(
     def stat(self) -> DataFrameStatFunctions:
         return self._stat
 
+    @overload
+    def collect(
+        self, *, statement_params: Optional[Dict[str, str]] = None, block: bool = True
```

Comment:
I'm curious, given that the default value of `block` is different though the function name is the same, does this mean `block` effectively becomes a required argument?